### PR TITLE
feat(PL-2466): add joy release validate command

### PR DESCRIPTION
**Pull request by davidmdm**
Result of running the new command against a local copy of the catalog with all releases using the generic chart with the values.cue schema:
```bash
go run ./cmd/joy --skip-version-check rel validate

Error: validating releases:
  - external-brokerage/qa: api.cronJobs.importer.pod.imageSufix: field not allowed
  - external-brokerage/staging: api.cronJobs.importer.pod.imageSufix: field not allowed
  - servicing-console/demo: frontend.container.targetPort: field not allowed
  - servicing-console/production: frontend.container.targetPort: field not allowed
  - servicing-console/qa: frontend.container.targetPort: field not allowed
  - servicing-console/staging: frontend.container.targetPort: field not allowed
  - tenants/demo: api.deployment.priorityClassName: field not allowed
  - tenants/production: api.deployment.priorityClassName: field not allowed
  - tenants/qa: api.deployment.priorityClassName: field not allowed
  - tenants/staging: api.deployment.priorityClassName: field not allowed
```

### CODE REVIEW

**Review by riclima — APPROVED** [review]

File: cmd/joy/release.go [hunk 1]
```diff
@@ -281,6 +285,69 @@ func NewReleaseRenderCmd() *cobra.Command {
...
+			}()
+
+			releaseFilter := func() filtering.Filter {
+				if len(args) == 0 {
+					return nil
+				}
+				return filtering.NewSpecificReleasesFilter(args)
+			}()
+
+			// Load catalog
```
Comment (silphid) [inline cmd/joy/release.go:312]: My favorite comment of all! I'm trying to remember the exact words a friend once told me...

File: internal/release/render/render.go [hunk 2]
```diff
@@ -50,36 +50,36 @@ func Render(ctx context.Context, params RenderOpts) error {
...
-		}
-	}
+type RenderReleaseOpts struct {
+	Release *v1alpha1.Release
+	Chart   *helm.Chart
+	CommonRenderOpts
+}
 
-	values, err := HydrateValues(release, environment, params.ValueMapping)
+func RenderRelease(ctx context.Context, params RenderReleaseOpts) error {
```
Comment (silphid) [inline internal/release/render/render.go:77]:
```suggestion
func RenderRelease(ctx context.Context, opts RenderReleaseOpts) error {
```

File: internal/release/validate/validate.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+	"github.com/davidmdm/x/xerr"
+
+	"github.com/nestoca/joy/api/v1alpha1"
+	"github.com/nestoca/joy/internal"
+	"github.com/nestoca/joy/internal/config"
+	"github.com/nestoca/joy/internal/helm"
+	"github.com/nestoca/joy/internal/release/render"
+)
+
+type ValidateParams struct {
```
Comment (silphid) [inline internal/release/validate/validate.go:25]: Let's try to stay consistent:
```suggestion
type ValidateOpts struct {
```

File: internal/release/validate/validate.go [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+		if err := ValidateRelease(ctx, validateParams); err != nil {
+			errs = append(errs, fmt.Errorf("%s/%s: %w", release.Name, release.Environment.Name, err))
+		}
+	}
+
+	return xerr.MultiErrOrderedFrom("validating releases", errs...)
+}
+
+type ValidateReleaseParams struct {
```
Comment (silphid) [inline internal/release/validate/validate.go:67]:
```suggestion
type ValidateReleaseOpts struct {
```

File: internal/release/validate/validate.go [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+	return nil
+}
+
+func AsErrorList[T error](list []T) []error {
+	result := make([]error, len(list))
+	for i, err := range list {
+		result[i] = err
+	}
+	return result
+}
```
Comment (silphid) [inline internal/release/validate/validate.go:133]: Wouldn't there be a way to handle that more gracefully in a single call to `xerr`?

**Review by silphid — APPROVED** [review]
Pre-approving, but please still review my comments...

File: internal/release/validate/validate.go (hunk 5 above)
Comment (davidmdm) [reply internal/release/validate/validate.go:136]: Not really. In cue if there's more than one error, the first is a parent and the others are children. Also it's a concrete type, and not of type error, so I need to convert them to a slice of error.

**Comment by j-martin** [comment]
> external-brokerage/qa: api.cronJobs.importer.pod.imageSufix: field not allowed

Not in this PR, but I think the `cue` spec is incorrect. All those jobs would fail without it.

https://github.com/nestoca/catalog/blob/e7e0d9f5d3b407a3336bb78f47ce99e496e13b3a/environments/production/releases/origination/accounts.yaml#L35-L73

Same thing with the `targetPort`.

**Comment by j-martin** [comment]
Another note:

You are right, `targetPort` in this context does nothing. I got mixed up with https://github.com/nestoca/catalog/blob/2893e91ff7be0e483b391d4590b65724999d9ae6/environments/production/releases/servicing/servicing-console.yaml#L43-L44

**Pull request merged**